src/Cutscenes.ts

```diff
@@ -1635,6 +1637,137 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
     }
 
     /**
+     * Cutscene for an Bubble attack in battle.
+     * 
+     * @param settings   Settings used for the cutscene.
+     * @param args   Settings for the routine.
+     */
+    public cutsceneBattleAttackBubble(settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+        const unitsize: number = this.EightBitter.unitsize;
+        const attackerName: string = args.attackerName;
+        const attacker: IThing = this.EightBitter.BattleMover.getThing(attackerName) as IThing;
+        const direction: number = attackerName === "player" ? 1 : -1;
+        const menu: IMenu = this.EightBitter.MenuGrapher.getMenu("BattleDisplayInitial") as IMenu;
+        const xPositions: number[] = new Array(4);
+        const yPositions: number[] = new Array(4);
+        const animateBubble: (x: number, y: number, i: number) => void = (x: number, y: number, i: number): void => {
+            if (i === 0) {
+                const bubbleLarge: IThing = this.EightBitter.ObjectMaker.make("BubbleLarge");
+                this.EightBitter.things.add(bubbleLarge, x, y);
+
+                this.EightBitter.TimeHandler.addEvent(
+                    (): void => {
+                        this.EightBitter.physics.killNormal(bubbleLarge);
+                    },
+                    4 * 24);
+            } else if (i === 1) {
+                const bubbleLarge: IThing = this.EightBitter.ObjectMaker.make("BubbleLarge");
+                const bubblesSmall: IThing[] = new Array(4);
```

Comment:
I'd prefer it if you avoided `Array` and other constructors unless absolutely necessary for performance. `= []` is fine at this scale.